### ISSUE TITLE
v1.0 - Required Features [Tracking]

### ISSUE THREAD

**Issue by drwasho** [OpenBazaar/OpenBazaar-Client#475]
This issue will be used to track subordinate issues required to get us to a finished v1.0 client.
- [x] #272 - Block node ... or @SamPatt will kill someone.
- [x] #467 - Search within Pages
- [x] #468 - Notifications (what's left)
- [x] #470 - Moderator functions
- [x] #471 - Transaction flow
- [x] #472 - Reputation functions
- [x] #473 - Blockchain ID

I think this captures all of them, and these issues are labelled `feature-complete requirement`. If there are any we've missed, let me know.

**Comment by drwasho** [comment]
In terms of priority, we absolutely need to finish #471 **asap**, which will enable some basic _alpha_-level testing. Please @jjeffryes make this your top priority. 

My own personal feeling is that moderator functions would be a close second, followed by search and reputation. Blockchain ID, notifications and block node being the last three in descending importance.

**Comment by drwasho** [comment]
### Week of 1st of February, 2016

The team has accomplished a lot the past week with the purchase flow functioning reliably and cleanly, which is a huge achievement considering how user friendly the whole process is. We've also got all the design documentation lined up to wire up the reputation functions.

We have the following tasks to complete to reach `feature complete` status:

**Notifications**
- [ ] New case to review (for moderators)
- [x] Dispute opened on a transaction
- [x] New message on a dispute
- [ ] Nudge buyer to release funds on a transaction (x days after order date)
- [ ] Moderator revoked their service from your store
- [x] Click through routing to relevant pages (i.e. Purchases, Sales, Cases etc)

**Transaction Flow**
- [ ] Build Dispute resolution controls for the Moderator
- [ ] Refund back to the Buyer

**Reputation Functions**
- [ ] Display the raw reputation summary of the user based off all trade summaries
- [ ] Show each individual rating + written feedback for all transactions

**Comment by SamPatt** [comment]
Also need to implement the NSFW filter. Neither stores nor individual listings tagged NSFW should be displayed unless the user has turned NFSW on in settings.

**Comment by jjeffryes** [comment]
We'll need DisplayNSFW: true/false added to the settings API. 

^ @cpacia

**Comment by drwasho** [comment]
### Week of 8th of February, 2016

A lot of positive progress has been made in the past week. A lot of features were added that weren't listed in the previous update (e.g. on-boarding and authentication) and existing components were polished, creating a much smoother experience. Now that we're closer to reaching `feature complete` status, the revised items left to do are:

**Notifications**
- [x] New case to review (for moderators)
- [ ] Nudge buyer to release funds on a transaction (x days after order date)
- [ ] Moderator revoked their service from your store

**Transaction Flow**
- [ ] Build Dispute resolution controls for the Moderator
  - Specifically, allowing the Moderator to control the % of funds that go to the Buyer and Vendor in a dispute
  - Control for the winning party in a dispute to release funds according to the Moderator's ruling (more sophisticated options like migrating to a different Moderator etc will be for future releases)
- [ ] Refund back to the Buyer
  - This step is for Vendors to cancel an order after the escrow address has been funded and refund or return those funds back to the Buyer; terminology will also need to be worked on

**Reputation Functions**
- [ ] Display the raw reputation summary of the user based off all trade summaries
- [x] Show each individual rating + written feedback for all transactions

**Comment by drwasho** [comment]
### Week of 23rd of February, 2016

We were in Amsterdam last week, so most of the team was out of action (but it was totally worth meeting with each other in person). 

Despite this, we managed to knock of a pretty important milestone related to reputation. Now when users visit a particular listing offered by the Vendor, you will be able to see the rating left by previous Buyers.

Rest of the issues:

**Notifications**
_These are looking more like stretch goals or features to be added after v1 release; not priority_
- [ ]  ~~Nudge buyer to release funds on a transaction (x days after order date)~~
- [ ]  ~~Moderator revoked their service from your store~~

**Transaction Flow**
_Still a v1 priority_
- [ ] Build Dispute resolution controls for the Moderator
  - Specifically, allowing the Moderator to control the % of funds that go to the Buyer and Vendor in a dispute
  - Control for the winning party in a dispute to release funds according to the Moderator's ruling (more sophisticated options like migrating to a different Moderator etc will be for future releases)
- [x] Refund back to the Buyer
  - This step is for Vendors to cancel an order after the escrow address has been funded and refund or return those funds back to the Buyer; terminology will also need to be worked on

**Reputation Functions**
_Still a v1 priority_
- [ ] ~~Display the raw reputation summary of the user based off all trade summaries~~

**Comment by SamPatt** [comment]
- [x] NSFW filter works. Requires both @jjeffryes and @cpacia

**Comment by drwasho** [comment]
@rmisio @cpacia @jjeffryes There are some bits of data that need to be displayed in the individual ratings for a listing, which I believe can pull from the rating message: 
1. GUID (no avatar; link out to try and reach their node when clicked)
2. TXID or the Bitcoin multisignature address from that transaction (links out to Blockchain.info)

Reasons: 
- **GUID** so users can try visit the purchaser and size them up
- **TXID or multisig address** to prove that the rating was associated with a real bitcoin transaction

**Comment by rmisio** [comment]
@cpacia Please let me know when the endpoint is updated to provide the data @drwasho is requesting and I'll update the UI.

**Comment by drwasho** [comment]
### Week of the 29th of February

We have a target release date for v1 on the testnet for March 1. After a big push this past week, the team has placed the finishing touches on the refund and dispute resolution controls. There are a couple of things we're going to try and finish before v1 is released on the mainnet (which we don't have a hard date for thus far).

If you want to help test out the refund and dispute controls, please follow this checklist:

```
1. Direct transaction
   -check funds paid out 

2. Moderated transaction 
   -check funds paid out

3. Direct transaction 
   -check refund successful

4. Moderated transaction 
   -check refund successful 

5. Moderated transaction - buyer dispute
   -check 100% payout to buyer successful

6. Moderated transaction - seller dispute
   -check 100% payout to seller successful

7. Moderated transaction - buyer dispute
   -check split payout works

8. Moderated transaction - seller dispute
   -check split payout works 
```

I've gone through it and found that all scenarios are fully working. We need some more people independently confirming this before we can close the issue.

**Transaction Flow**
_Still a v1 priority_
- [x] Build Dispute resolution controls for the Moderator
  - Specifically, allowing the Moderator to control the % of funds that go to the Buyer and Vendor in a dispute
  - Control for the winning party in a dispute to release funds according to the Moderator's ruling (more sophisticated options like migrating to a different Moderator etc will be for future releases)

**Comment by drwasho** [comment]
We're ready.
